### PR TITLE
Interactive "compose down" when deploying too many services

### LINKED ISSUE #665: Create interactive prompt "compose down y/n?" when trying to deploy 2nd service

### PR DESCRIPTION
**Pull request by commit111**
Fixes #665 

New Behavior:
- When a user tries to run `compose up`, but `maximum number of projects` is reached, we ask if they want us to `compose down` on a project
   - If yes, we run `compose down` for them, and then print a hint for them to retry `compose up` 
   - If no, we print a hint for them to try `compose down` on their own

Details:
- `InteractiveComposeDown()` to create prompt logic
- `SetProjectName()` in PlaygroundClient
  - This function is used to set the client’s project name to the correct one for deletion;
  otherwise, the client would delete the project intended for deployment 
  - Changes are reflected in Client interface, MockClient, BaseClient files
- Added a field `projectName` to `ComposeDown()` to specify project to delete

### CODE REVIEW

File: src/cmd/cli/command/commands.go [hunk 1]
```diff
@@ -56,6 +56,27 @@ func prettyError(err error) error {
 	return err
 }
 
+func interactiveComposeDown(ctx context.Context, c cliClient.Client, projectName string) error {
```
Comment (lionello) [inline src/cmd/cli/command/commands.go:59]: Shall we move this function to `composeDown.go`? This file is pretty large already

File: src/cmd/cli/command/commands.go [hunk 2]
```diff
@@ -56,6 +56,27 @@ func prettyError(err error) error {
...
+	var wantComposeDown bool
+	err := survey.AskOne(&survey.Confirm{
+		Message: "Run 'compose down' to deactivate project: " + projectName + "?",
+	}, &wantComposeDown)
+
+	if err != nil {
+		return err
```
Comment (lionello) [inline src/cmd/cli/command/commands.go:66]: nobody checks this `err`, so either check in the caller (and `term.Debug`?) or print something here.

**Review by lionello — APPROVED** [review]
looks good! Just left a suggestion to move the new function to another file.

**Review by nullfunc — APPROVED** [review]

File: src/cmd/cli/command/commands.go (hunk 1 above)
Comment (commit111) [reply src/cmd/cli/command/commands.go]: Ok, done! There are a few adjustments I made to my previous code to make it work (i.e. the hints are now done by the caller)

File: src/cmd/cli/command/commands.go [hunk 3]
```diff
@@ -89,8 +89,14 @@ func Execute(ctx context.Context) error {
 			projectName := "<name>"
```
Comment (lionello) [inline src/cmd/cli/command/commands.go:89]: Let's move this if into `makeComposeUpCmd` https://github.com/DefangLabs/defang/blob/b3451afbf239738c6ae1a71b55feb004c8ac69fa/src/cmd/cli/command/compose.go#L41

File: src/cmd/cli/command/commands.go [hunk 4]
```diff
@@ -89,8 +89,14 @@ func Execute(ctx context.Context) error {
...
 			if resp, err := client.GetServices(ctx); err == nil {
 				projectName = resp.Project
+				err := cli.InteractiveComposeDown(ctx, client, projectName)
```
Comment (lionello) [inline src/cmd/cli/command/commands.go:92]: Only prompt if `!nonInteractive`

**Review by lionello — CHANGES_REQUESTED** [review]

File: src/cmd/cli/command/commands.go (hunk 3 above)
Comment (defanguserxoxo) [reply src/cmd/cli/command/commands.go]: ok, it's updated now

Comment (commit111) [reply src/cmd/cli/command/commands.go]: ok, it's updated now

File: src/cmd/cli/command/compose.go [hunk 5]
```diff
@@ -50,11 +50,22 @@ func makeComposeUpCmd() *cobra.Command {
...
 			deploy, project, err := cli.ComposeUp(cmd.Context(), client, force, mode.Value())
+
 			if err != nil {
-				if !errors.Is(err, types.ErrComposeFileNotFound) {
-					return err
+				if !nonInteractive && strings.Contains(err.Error(), "maximum number of projects") {
+					if resp, err2 := client.GetServices(cmd.Context()); err2 == nil {
+						term.Error("Error:", prettyError(err))
+						if err := cli.InteractiveComposeDown(cmd.Context(), client, resp.Project); err != nil {
+							printDefangHint("To deactivate a project, do:", "compose down --project-name "+resp.Project)
```
Comment (lionello) [inline src/cmd/cli/command/compose.go:59]:
```suggestion
						if _, err := cli.InteractiveComposeDown(cmd.Context(), client, resp.Project); err != nil {
							term.Debug("ComposeDown failed:", err)
							printDefangHint("To deactivate a project, do:", "compose down --project-name "+resp.Project)
```

File: src/pkg/cli/composeDown.go [hunk 6]
```diff
@@ -32,3 +39,24 @@ func ComposeDown(ctx context.Context, client client.Client, names ...string) (ty
...
+	}
+
+	if !wantComposeDown {
+		return ErrDoNotComposeDown
+	} else {
+		term.Info("Deactivating project " + projectName)
+		ComposeDown(ctx, c, projectName)
+		return nil
+	}
+}
```
Comment (lionello) [inline src/pkg/cli/composeDown.go:62]:
```suggestion
func InteractiveComposeDown(ctx context.Context, c client.Client, projectName string) (types.ETag, error) {
	var wantComposeDown bool
	err := survey.AskOne(&survey.Confirm{
		Message: "Run 'compose down' to deactivate project: " + projectName + "?",
	}, &wantComposeDown)

	if err != nil {
		return "", err
	}

	if !wantComposeDown {
		return "", ErrDoNotComposeDown
	}
	
	term.Info("Deactivating project " + projectName)
	return ComposeDown(ctx, c, projectName)
}
```

**Review by lionello — APPROVED** [review]
Just two nits

**Pull request merged**